app/enterprise/2.3.x/plugins/oidc-curity.md

```diff
@@ -0,0 +1,209 @@
+* An introspection endpoint configured with the [Token Procedure Approach][curity-phantom-token-introspection]
+
+### Configure Kong
+
+The Kong OpenID Connect plugin is configured to introspect an incoming opaque access token and in return receive a JWT in the introspection response from the Curity Identity Server. The plugin is enabled for a [service or a route][kong-add-service].
+
+As part of the introspection, the OpenID Connect plugin also has the ability to validate that required scopes are available in the introspected token. Access to the requested API are denied if the correct scopes are missing.
+
+If access is granted, the JWT from the introspection response is added to a header and forwarded to the upstream API where it can be consumed.
+
+#### Create a service
+
+Create a service that can be used to test the integration.
+
+```bash
+curl -i -X POST http://kong:8001/services/ \
+  --data name="httpbin" \
+  --data protocol="http" \
+  --data url="http://httpbin.org"
+```
+
+#### Create a route
+
+Add a route to the service.
+
+```bash
+curl -i -X POST http://kong:8001/services/httpbin/routes \
+  --data "paths[]=/httpbin"
+```
+
+#### Configure the plugin
+
+The Kong OpenID Connect plugin is enabled for the previously created service. In the example below, the `openid` scope is required in order for access to be granted. As noted by the `config.upstream_headers_claims` configuration, the plugin looks for the `JWT` (the phantom token) claim in the introspection response. The `config.upstream_headers_names` configuration extracts the `JWT` from the introspection response and adds it to a `phantom_token` header in the call to the upstream API. 
+
+```bash
+curl -X POST http://kong:8001/services/httpbin/plugins \
+--data name="openid-connect" \
+--data config.issuer="https://idsvr.example.com/oauth/v2/oauth-anonymous" \
+--data config.client_id="gateway-client" \
+--data config.client_secret="Password1" \
+--data config.scopes_required="openid" \
+--data config.hide_credentials="true" \
+--data config.upstream_headers_claims="phantom_token" \
+--data config.upstream_headers_names="phantom_token" \
+--data config.auth_methods="introspection"
+```
+
+Parameter | Description | Example | Required for integration
+--- |--- |--- |---
+`config.issuer` | Used for discovery. Kong appends `/.well-known/openid-configuration`. Should be set to the `realm` or `iss` if no discovery endpoint is available. | `https://idsvr.example.com/oauth/v2/oauth-anonymous` | Yes
+`config.client_id` | The ID of a client with the introspection capability | `gateway-client`| Yes
+`config.client_secret` | Secret of the client used for introspection| `Password1` | Yes
+`config.scopes_required`| Optional scopes required in introspection result for authorization. | `openid email records_read`  | No
+`config.hide_credentials` | Boolean value. This will prevent the incoming Access Token from being forwarded to the upstream API. |`true` | No
```

Comment:
You might need to set `config.upstream_access_token_header=` (aka nil) too because that might add it back. As this client can receive tokens with multiple names and ways, the hide_credential just removes that, but `config.upstream_access_token_header` is set by default to `authorization:bearer`, so that in this case adds it back.

Reply:
Added clarification